staging/src/k8s.io/apiserver/pkg/cel/ip_addr.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"fmt"
+	"net/netip"
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+)
+
+// IPAddr provides a CEL representation of an IP address.
+type IPAddr struct {
+	*netip.Addr
+}
+
+var (
+	IPAddrObject    = decls.NewObjectType("kubernetes.IPAddr")
+	ipAddrTypeValue = types.NewTypeValue("kubernetes.IPAddr")
+	IPAddrType      = cel.ObjectType("kubernetes.IPAddr")
+)
+
+// ConvertToNative implements ref.Val.ConvertToNative.
+func (d IPAddr) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
```

Comment:
I'm not sure why there would be conversion to a `net.IP` since this library is intended to be used only within the CEL validation, meaning, IIUC, built in types/anything that already exists with `net.IP` compatibility wouldn't be anywhere near this code

Reply:
ah, sorry, I haven't done anything with CEL extensions and was just kind of guessing at what some of this is for. :slightly_smiling_face: I wasn't sure if this function affected what types you could use for the field when generating structs from a CRD...

Reply:
Nah, nothing to do with the generation, this can be super simple and isolated I think.